game_event/views.py

```diff
@@ -63,3 +65,52 @@ def game_events(request):
         game_events = game_events.filter(time__gte=current_date)
 
     return render(request, 'game-events.html', {'game_events': game_events})
+
+
+def game_event(request, id):
+    event = get_object_or_404(GameEvent, pk=id)
+    player = Player.objects.get(user=request.user)
+    event_players = GameEventPlayer.objects.filter(game_event=event)
+    event_players = [{'first_name': entry.player.user.first_name, 'last_name': entry.player.user.last_name,
+                     'brings_ball': entry.ball_responsible} for entry in event_players]
+    in_event = False
+    if GameEventPlayer.objects.filter(game_event=event, player=player).exists():
+        in_event = True
+    context = {'id': event.id, 'time': event.time, 'level_of_game': event.level_of_game, 'min_number_of_players':
+               event.min_number_of_players, 'max_number_of_players': event.max_number_of_players,
+               'court': event.court.city, 'neighborhood': event.court.neighborhood,
+               'ball_game': event.ball_game, 'in_event': in_event, 'event_players': event_players}
+
+    return render(request, 'game_event/game-event.html', context)
+
+
+def join_event(request, id):
+    get_object_or_404(GameEvent, pk=id)
```

Comment:
Is that for checking that the event exists?

Reply:
yes

Reply:
What will happen if you don't do it?

Reply:
I will need to use instead 'GameEvent.objects.get(id=id)' , but after I need to check that the event exists.
'get_object_or_404(GameEvent, pk=id)' does it for me.

Reply:
What will happen if you try to join an event that doesn't exist?

The code smells suspicious for two main reasons:
1. You have a `get` call but you don't use the result.
2. If it's just for validating then you have a race condition here: what happens if the event is removed between your check and the `render` call? You are then left with the same situation as you were without having the check.

It's sometimes counter-intuitive, but in most cases it's safer to just try to do what you're trying to do, and process the failure later, rather than checking beforehand that you should be able to do it, as things can change between the check and the doing (asking for forgiveness instead of asking for permission).

Practically, it means putting your doing inside a `try` block and processing the failures inside an `except` block.

Reply:
I didn't think I could get into a race condition, now I get the point.
by adding the ```try``` ```except``` how will it resolve the race condition?

Reply:
The race is between the time that you check the existence and the time that you try to access. If you don't check it exists and just try to access it then there's nothing to race against. It will either succeed or fail. If it fail, you address it on the except block.

Reply:
You should address it in the exact same way that you would before.

Reply:
do I add empty content if I don't have a game event?

Reply:
Up to you. Just make it coherent.